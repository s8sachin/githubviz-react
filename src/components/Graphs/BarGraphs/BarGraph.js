import React, { Component } from 'react';
import { XYPlot,VerticalBarSeries,XAxis,YAxis, VerticalGridLines, HorizontalGridLines,Hint,LineMarkSeries
  } from 'react-vis';
  import { Grid, Row,Col} from 'react-bootstrap';

const data = [
    {x: 0, y: 8},
    {x: 1, y: 5},
    {x: 2, y: 4},
    {x: 3, y: 9},
    {x: 4, y: 1},
    {x: 5, y: 7},
    {x: 6, y: 6},
    {x: 7, y: 3},
    {x: 8, y: 2},
    {x: 9, y: 0}
  ];

class BarGraph extends Component {
    state = {
        value: false
      }
  render() {
    const {value} = this.state;
    return (
        <div>
             <Grid>
                 <Row className="show-grid">
                    <Col xs={12} md={12}>
                     <div>
                        <XYPlot height={300} width={500} color="orange" stroke="black"  xType="ordinal">
                        <VerticalGridLines />
                        <HorizontalGridLines />
                        <XAxis />
                        <YAxis />
                        <VerticalBarSeries data={data}
                            onValueMouseOver={v => this.setState({ value: v })}
                            onSeriesMouseOut={v => this.setState({ value: false })} />
                        {value && <Hint value={value} />}
                        </XYPlot>
                    </div>
                 </Col>
             </Row>
            </Grid>
        </div>
    );
  }
}

export default BarGraph;
