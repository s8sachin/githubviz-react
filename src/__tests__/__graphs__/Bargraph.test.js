import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import BarGraph from '../../components/Graphs/BarGraphs/BarGraph';

configure({ adapter: new Adapter() });

describe('BarGraph', () => {
it('should mount Header component', () => {
    const wrapper = shallow(<BarGraph />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})