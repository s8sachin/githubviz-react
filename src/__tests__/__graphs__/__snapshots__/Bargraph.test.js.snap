// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarGraph should mount Header component 1`] = `
<div>
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Row
      bsClass="row"
      className="show-grid"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={12}
        xs={12}
      >
        <div>
          <XYPlot
            className=""
            color="orange"
            height={300}
            stroke="black"
            width={500}
            xType="ordinal"
          >
            <VerticalGridLines
              attr="x"
              direction="vertical"
            />
            <HorizontalGridLines
              attr="y"
              direction="horizontal"
            />
            <XAxis
              attr="x"
              attrAxis="y"
              orientation="bottom"
            />
            <YAxis
              attr="y"
              attrAxis="x"
              orientation="left"
            />
            <VerticalBarSeries
              className=""
              data={
                Array [
                  Object {
                    "x": 0,
                    "y": 8,
                  },
                  Object {
                    "x": 1,
                    "y": 5,
                  },
                  Object {
                    "x": 2,
                    "y": 4,
                  },
                  Object {
                    "x": 3,
                    "y": 9,
                  },
                  Object {
                    "x": 4,
                    "y": 1,
                  },
                  Object {
                    "x": 5,
                    "y": 7,
                  },
                  Object {
                    "x": 6,
                    "y": 6,
                  },
                  Object {
                    "x": 7,
                    "y": 3,
                  },
                  Object {
                    "x": 8,
                    "y": 2,
                  },
                  Object {
                    "x": 9,
                    "y": 0,
                  },
                ]
              }
              onSeriesMouseOut={[Function]}
              onValueMouseOver={[Function]}
              stack={false}
              style={Object {}}
            />
          </XYPlot>
        </div>
      </Col>
    </Row>
  </Grid>
</div>
`;
